_posts/2024-12-27-this-week-in-rails.markdown

```diff
@@ -0,0 +1,164 @@
+to help your app run efficiently and safely on the open Internet.
+
+This change configures the use of Thruster in the Dockerfile by default, starting with Rails 8.
+
+[Add support for SQLite3 full-text search and other virtual tables](https://github.com/rails/rails/pull/52354)  
+SQLite3 full-text search and other virtual tables are now supported in Rails.
+Previously, adding SQLite3 virtual tables messed up `schema.rb`, but with this change, virtual tables can safely be added using `create_virtual_table`.
+
+[Add Solid Cache](https://github.com/rails/rails/pull/52790)  
+Solid Cache will be the new default caching backend for production deployments out of the box in Rails 8.
+
+[Add Solid Queue](https://github.com/rails/rails/pull/52804)  
+Configure Solid Queue as the default Active Job backend alongside Solid Cache. Both can be skipped with `--skip-solid`.
+
+[Silence healthcheck requests from the log](https://github.com/rails/rails/pull/52789)  
+Add `Rails::Rack::SilenceRequest` middleware and use it via `config.silence_healthcheck_path = path` to silence requests to "/up".
+This prevents the Kamal-required healthchecks from clogging up the production logs.
+
+[Add Parameters#expect to safely filter and require params](https://github.com/rails/rails/pull/51674)  
+This PR adds a new way to handle params giving more control over what you expect to receive in your controller actions.
+```ruby
+# Before
+params.require(:table).permit(:attr)
+
+# After
+params.expect(table: [ :attr ])
+```
+
+[Tidy up the error pages](https://github.com/rails/rails/pull/53045)  
+The error pages built into Rails have been updated, here's a preview of the new look:
+<img width="1512" alt="Screenshot 2024-09-26 at 10 46 56 AM" src="https://github.com/user-attachments/assets/225dbc1f-309e-4f31-9da7-106f7d312424">
+
+[Support loading SQLite3 extensions](https://github.com/rails/rails/pull/53827)  
+The *sqlite3* gem v2.4.0 introduces support for loading extensions passed as a kwarg to *Database.new*. This PR leverages that feature to allow configuration of extensions in the *config/database.yml* file using either filesystem paths or the names of modules that respond to *to_path* method.
+
+[Add ActiveSupport::Testing::NotificationAssertions test helper module](https://github.com/rails/rails/pull/53065)  
+With this addition, we can now use various test helper methods for notification assertions:
+
+```ruby
+assert_notification("post.submitted", title: "Cool Post") do
+  post.submit(title: "Cool Post") # => emits matching notification
+end
+
+assert_notifications_count("post.submitted", 1) do
+  post.submit(title: "Cool Post")
+end
+
+assert_no_notifications("post.submitted") do
+  post.destroy
+end
+
+notifications = capture_notifications("post.submitted") do
+  post.submit(title: "Cool Post") # => emits matching notification
+end
+```
+
+[Migrate `ActiveRecord::Normalization` to Active Model](https://github.com/rails/rails/pull/53887)  
```

Comment:
It looks like this pull request has been reverted and reopened, but it's not yet merged: https://github.com/rails/rails/pull/53955.

Reply:
Thanks for spotting it. I will remove it.